### ISSUE TITLE
Steps misaligned

### ISSUE THREAD

**Issue by korry73** [SashaCX75/AmazFit_Watchface_Editor#41]
Hi, great editor, just one thing thing that takes away my sleep. I want to create a simple watchface for my GTS with big numbers for time and steps (some other infos are displayed using smaller numbers). Everything works, but the steps are not aligned vertically, some of the numbers appear ~2 pixels above the others. What is strange, it is also position dependent. For example, I have 3882 steps so far for today and the second "8" appears a bit higher than the first "8". Could you please have a look? The numbers from 0012 to 0021 have same height of 75 pixels.

This is my json:

{
  "Info": {
    "DeviceId": 46
  },
  "Background": {
    "Image": {
      "X": 0,
      "Y": 0,
      "ImageIndex": 0
    },
    "Preview": {
      "X": 0,
      "Y": 0,
      "ImageIndex": 76
    }
  },
  "Time": {
    "Hours": {
      "Tens": {
        "X": 20,
        "Y": 128,
        "ImageIndex": 2,
        "ImagesCount": 10
      },
      "Ones": {
        "X": 85,
        "Y": 128,
        "ImageIndex": 2,
        "ImagesCount": 10
      }
    },
    "Minutes": {
      "Tens": {
        "X": 200,
        "Y": 128,
        "ImageIndex": 2,
        "ImagesCount": 10
      },
      "Ones": {
        "X": 270,
        "Y": 128,
        "ImageIndex": 2,
        "ImagesCount": 10
      }
    },
    "Delimiter": {
      "X": 158,
      "Y": 117,
      "ImageIndex": 69
    }
  },
  "Activity": {
    "Calories": {
      "TopLeftX": 50,
      "TopLeftY": 362,
      "BottomRightX": 166,
      "BottomRightY": 400,
      "Alignment": "CenterLeft",
      "Spacing": 4,
      "ImageIndex": 32,
      "ImagesCount": 10
    },
    "Distance": {
      "Number": {
        "TopLeftX": 206,
        "TopLeftY": 362,
        "BottomRightX": 330,
        "BottomRightY": 400,
        "Alignment": "CenterLeft",
        "Spacing": 4,
        "ImageIndex": 32,
        "ImagesCount": 10
      },
      "SuffixImageIndex": 76,
      "DecimalPointImageIndex": 71
    },
    "Steps": {
      "Step": {
        "TopLeftX": 50,
        "TopLeftY": 248,
        "BottomRightX": 330,
        "BottomRightY": 5,
        "Alignment": "CenterLeft",
        "Spacing": 8,
        "ImageIndex": 12,
        "ImagesCount": 10
      }
    }
  },
  "Date": {
    "MonthAndDay": {
      "Separate": {
        "Month": {
          "TopLeftX": 96,
          "TopLeftY": 52,
          "BottomRightX": 0,
          "BottomRightY": 0,
          "Alignment": "TopLeft",
          "Spacing": 6,
          "ImageIndex": 22,
          "ImagesCount": 10
        },
        "Day": {
          "TopLeftX": 20,
          "TopLeftY": 52,
          "BottomRightX": 0,
          "BottomRightY": 0,
          "Alignment": "TopLeft",
          "Spacing": 6,
          "ImageIndex": 22,
          "ImagesCount": 10
        }
      },
      "TwoDigitsMonth": true,
      "TwoDigitsDay": true
    },
    "WeekDay": {
      "X": 168,
      "Y": 52,
      "ImageIndex": 52,
      "ImagesCount": 7
    }
  },
  "Weather": {
    "Temperature": {
      "Current": {
        "TopLeftX": 250,
        "TopLeftY": 52,
        "BottomRightX": 330,
        "BottomRightY": 86,
        "Alignment": "TopRight",
        "Spacing": 5,
        "ImageIndex": 42,
        "ImagesCount": 10
      },
      "Symbols": {
        "Unknown0800": 0,
        "MinusImageIndex": 73,
        "DegreesImageIndex": 72,
        "NoDataImageIndex": 76
      }
    }
  },
  "Status": {
    "Alarm": {
      "Coordinates": {
        "X": 60,
        "Y": 4
      },
      "ImageIndexOn": 74,
      "ImageIndexOff": 76
    },
    "DoNotDisturb": {
      "Coordinates": {
        "X": 100,
        "Y": 4
      },
      "ImageIndexOn": 75,
      "ImageIndexOff": 76
    }
  },
  "Battery": {
    "Images": {
      "X": 250,
      "Y": 2,
      "ImageIndex": 59,
      "ImagesCount": 10
    }
  }
}

**Comment by SashaCX75** [comment]
@korry73 
The same numbers cannot be displayed at different heights. Different numbers can be displayed at different heights if they are drawn at different heights inside the picture.

**Comment by korry73** [comment]
Please have a look at https://zornica.info/numbers2.jpg
The second "5" is somewhat scaled and appears smaller compared to the first one. But it is only an example. Most of the last digits appear smaller compared to the second-last. All numbers are here: https://zornica.info/numbers.zip. I guess it may be related to the quite big height of 75 pixels I would like to use, my numbers with a smaller height for date, distance and calories do not have the problem.

**Comment by SashaCX75** [comment]
@korry73 
По фото вроде все нормально. Для проверки можно сделать на картинках с цифрами фон разного цвета. Тогда четко будет видно если цифры будут смещены.

From the photo everything seems to be fine. For verification, you can make a background of different colors on the pictures with numbers. Then it will be clearly visible if the numbers are shifted.

**Comment by korry73** [comment]
I have replaced all numbers by rectangles 50 x 100 pixels and they were aligned correctly. Then I have used numbers from another watchface, all 65 x 75 pixeld, and oddly, they look misaligned. Really strange. I am not a programmer, but I believe that a picture is displayed using given coordinates. It is also wired, that numbers that are smaller in size are displayed correctly. But anyway... we can close the issue. I have played with the numbers, so mine are not same size now, some are 74 pixels high and some even 73, but it looks better on the device. Hard to believe, but this is the truth.
I am Bulgarian and I understand Russian, it is just much easier for me to write in English :)

**Comment by SashaCX75** [comment]
@korry73 
I meant that on the same numbers that are used in the dial, the background should be colored instead of the transparent one. Then it will be seen for sure that the pictures are displayed at different heights or, due to the peculiarities of the font, it visually seems that the numbers are at a different level.

I wrote in Russian because in the photo the designation of the days of the week looked like Russian, and when translated into English, sometimes the meaning is lost.

**Comment by korry73** [comment]
Yep, in my test I have used just green rectangles w/o transparency for all numbers and they were aligned, so it is maybe the font, that makes it look misaligned. Then I have played with the numbers (I use Inkscape to create the images), some now have a height of 75, some of 74, some of 73 pixels and it looks better on the watch - do not ask me why, this is strange to me. Anyway, thanks for your support and for the great app.
